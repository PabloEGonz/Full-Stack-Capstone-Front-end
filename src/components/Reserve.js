import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

// import { useNavigate } from 'react-router-dom';
import { createReserve, getReservations } from '../redux/reservations/reservationsSlice';
import { getCars } from '../redux/cars/carsSlice';

const Reservation = () => {
  const cars = useSelector((state) => state.cars.cars);
  // const user = useSelector((state) => state.user.id);
  const userId = useSelector((state) => state.user.id);
  console.log(userId);

  // const navigate = useNavigate();

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCars());
    dispatch(getReservations(userId));
  }, [dispatch]);

  const [reserve, setReserve] = useState({
    user_id: userId,
    reservation_date: '10-11-2023',
    due_date: '',
    service_fee: '',
    car_id: '',
  });

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setReserve((prevState) => ({
      ...prevState,
      [name]: value,
    }));
  };

  const submit = () => {
    console.log(reserve);
    dispatch(createReserve(reserve));

    // e.preventDefault();
    // const {
    //   reservationDate, dueDate, serviceFee, car,
    // } = reserve;

    // if (!reservationDate || !dueDate || !serviceFee || !car) {
    //   alert('Please fill in all the required fields.');
    //   return;
    // }

    // try {
    //   // Dispatch the createReserve action to make the reservation request
    //   await dispatch(createReserve({ ...reserve, userId: user }));
    //   alert('Reservation created successfully!');
    //   navigate('/reservations'); // Redirect to the reservation page after successful reservation
    // } catch (error) {
    //   alert('Error occurred while making a reservation.');
    //   console.error(error);
    // }
  };

  return (
    <div>
      <h2>Create Reservation</h2>
      <form>
        <div>
          <input
            type="date"
            id="reservationDate"
            name="reservation_date"
            value={reserve.reservation_date}
            onChange={handleInputChange}
            placeholder="Reservation Date"
            required
          />
        </div>
        <div>
          <input
            type="date"
            id="dueDate"
            name="due_date"
            value={reserve.due_date}
            onChange={handleInputChange}
            placeholder="Due Date"
            required
          />
        </div>
        <div>
          <input
            type="number"
            id="serviceFee"
            name="service_fee"
            value={reserve.service_fee}
            onChange={handleInputChange}
            placeholder="Service Fee"
            required
          />
        </div>
        <div>
          <select
            id="car"
            name="car_id"
            value={reserve.car_id}
            onChange={handleInputChange}
            required
          >
            <option value="">Select a car</option>
            {cars.map((car) => (
              <option key={car.id} value={car.id}>
                {car.name}
              </option>
            ))}
          </select>
        </div>
        <button type="button" onClick={submit}>Create Reservation</button>
      </form>
    </div>
  );
};

export default Reservation;
