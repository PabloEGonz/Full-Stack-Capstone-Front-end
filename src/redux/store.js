import { configureStore } from '@reduxjs/toolkit';
import userReducer from './users/usersSlice';
import reservationsSlice from './reservations/reservationsSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    reservations: reservationsSlice,
  },
});
export default store;
