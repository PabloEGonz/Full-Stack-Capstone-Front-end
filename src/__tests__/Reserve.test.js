import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Reserve from '../components/Reserve';
import store from '../redux/store';

const cars = [
  {
    id: 1,
    name: 'Car 1',
    image: 'https://images.unsplash.com/photo-1542281286-9e0a16bb7366',
  },
  {
    id: 2,
    name: 'Car 2',
    image: 'https://images.unsplash.com/photo-1542281286-9e0a16bb7366',
  },
];

const user = 1;

it('Reserve component should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <Reserve cars={cars} user={user} />
        </Provider>
      </Router>
    </React.StrictMode>,
  );

  expect(result).toMatchSnapshot();
});
