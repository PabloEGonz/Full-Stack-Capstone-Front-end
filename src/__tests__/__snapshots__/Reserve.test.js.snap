// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarReservation component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-5"
      >
        <h2>
          Create Reservation
        </h2>
        <form
          class="reservation-form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="reservationDate"
            >
              Reservation Date
            </label>
            <input
              class="form-control"
              id="reservationDate"
              name="reservation_date"
              required=""
              type="date"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="dueDate"
            >
              Due Date
            </label>
            <input
              class="form-control"
              id="dueDate"
              name="due_date"
              required=""
              type="date"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="serviceFee"
            >
              Service Fee
            </label>
            <div
              class="input-group"
            >
              <span
                class="input-group-text"
              >
                $
              </span>
              <input
                class="form-control"
                id="serviceFee"
                name="service_fee"
                required=""
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="car"
            >
              Select a Car
            </label>
            <select
              class="form-select"
              id="car"
              name="car_id"
              required=""
            >
              <option
                value=""
              >
                Select a car
              </option>
            </select>
          </div>
          <button
            class="btn btn-primary"
            type="button"
          >
            Create Reservation
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-5"
    >
      <h2>
        Create Reservation
      </h2>
      <form
        class="reservation-form"
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="reservationDate"
          >
            Reservation Date
          </label>
          <input
            class="form-control"
            id="reservationDate"
            name="reservation_date"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="dueDate"
          >
            Due Date
          </label>
          <input
            class="form-control"
            id="dueDate"
            name="due_date"
            required=""
            type="date"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="serviceFee"
          >
            Service Fee
          </label>
          <div
            class="input-group"
          >
            <span
              class="input-group-text"
            >
              $
            </span>
            <input
              class="form-control"
              id="serviceFee"
              name="service_fee"
              required=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="car"
          >
            Select a Car
          </label>
          <select
            class="form-select"
            id="car"
            name="car_id"
            required=""
          >
            <option
              value=""
            >
              Select a car
            </option>
          </select>
        </div>
        <button
          class="btn btn-primary"
          type="button"
        >
          Create Reservation
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Reservation component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mt-5"
      >
        <div
          class="row justify-content-center"
        >
          <div
            class="col-md-6"
          >
            <div
              class="alert alert-danger"
              role="alert"
            >
              You need to log-in or sign-in to continue
            </div>
            <a
              class="btn"
              href="/session#login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mt-5"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-md-6"
        >
          <div
            class="alert alert-danger"
            role="alert"
          >
            You need to log-in or sign-in to continue
          </div>
          <a
            class="btn"
            href="/session#login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Reserve component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loading"
      >
        <div
          class="spinner"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loading"
    >
      <div
        class="spinner"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
